src/store/assets/tokens.ts

```diff
@@ -58,25 +62,84 @@ export const assetsState = atom(get =>
   get(waitForAll(transformAssets(get(currentAccountDataState)?.balances).map(assetItemState)))
 );
 
+export const assetsUnanchoredState = atom(get =>
+  get(
+    waitForAll(
+      transformAssets(get(currentAccountDataState)?.unanchoredBalances).map(assetItemState)
+    )
+  )
+);
+
 export const transferableAssetsState = atom(get =>
   get(assetsState)?.filter(asset => asset.canTransfer)
 );
 
-export const fungibleTokensState = atom(get => {
-  const assets = get(assetsState);
-  return assets?.filter(asset => asset.type === 'ft');
-});
+const calculateBalanceByType = atomFamily((assetType: string) =>
+  atom(get => {
+    const assets: AssetWithMeta[] = get(assetsState);
+    const assetMap = new Map();
+    // Merge both balances (unanchored and anchored)
+    assets.forEach(
+      asset =>
+        asset.type === assetType &&
+        assetMap.set(asset.subtitle, { ...asset, ...{ subBalance: '0' } })
+    );
+    const unanchoredAssets: AssetWithMeta[] = get(assetsUnanchoredState);
+    unanchoredAssets.forEach(asset => {
+      if (asset.type !== assetType) return;
+      if (!assetMap.has(asset.subtitle)) {
+        asset.subBalance = asset.balance;
+        asset.balance = '0';
+        assetMap.set(asset.subtitle, asset);
+      } else {
+        assetMap.get(asset.subtitle).subBalance = asset.balance;
+      }
+    });
+    return [...assetMap.values()];
+  })
+);
+
+export const fungibleTokensState = calculateBalanceByType('ft');
+
 export const nonFungibleTokensState = atom(get => {
-  const assets = get(assetsState);
-  return assets?.filter(asset => asset.type !== 'nft');
+  const balances = get(currentAccountDataState)?.balances;
+  const unanchoredBalances = get(currentAccountDataState)?.unanchoredBalances;
+  const anchoredNfts = balances?.non_fungible_tokens || {};
+  const unanchoredNfts = unanchoredBalances?.non_fungible_tokens || {};
+  const noCollectibles =
+    Object.keys(anchoredNfts).length === 0 && Object.keys(unanchoredNfts).length === 0;
+
+  if (noCollectibles) return [];
+
+  const assets = Object.keys(anchoredNfts);
+  const assetMap = new Map();
+  // Merge both balances (unanchored and anchored)
+  assets.forEach(asset =>
+    assetMap.set(asset, { ...anchoredNfts[asset], ...{ subCount: '0', key: asset } })
+  );
+
+  Object.keys(unanchoredNfts).forEach(key => {
+    const asset = unanchoredNfts[key];
+    if (!assetMap.has(key)) {
+      assetMap.set(key, { ...asset, ...{ subCount: asset.count, count: '0', key } });
+    } else {
+      assetMap.get(key).subCount = asset.count;
+    }
+  });
+
+  return [...assetMap.values()];
 });
+
 export const stxTokenState = atom(get => {
   const balances = get(accountBalancesState);
-  if (!balances || balances.stx.balance === '0') return;
+  const unanchoredBalances = get(accountUnanchoredBalancesState);
+  if (!balances || !unanchoredBalances) return;
+  if (balances.stx.balance === '0' && unanchoredBalances.stx.balance === '0') return;
```

Comment:
Not necessarily to be fixed here, but this is extremely problematic imo. `'0' !== '0.00'`. Strings shouldn't be used to compare numerical values.

`BigNumber.isEqualTo`

Reply:
Made issue here to address this https://github.com/blockstack/stacks-wallet-web/issues/1519